client/src/main/java/com/paypal/selion/platform/html/AbstractElement.java

```diff
@@ -485,6 +485,9 @@ public void click(Object... expected) {
         if (expected == null || expected.length == 0) {
             return;
         }
+        if (parent != null) {
+            WebDriverWaitUtils.waitUntilPageIsLoaded(parent.getCurrentPage());
+        }
```

Comment:
Re: lines 510 to 512 in this this file, when `expected` contains a `WebPage` do you need to `waitUntilPageIsLoaded` on the `WegPage` before calling `waitUntilPageIsValidated`?